src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,668 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    public sealed partial class ConfigurationBindingSourceGenerator
+    {
+        private sealed partial class Emitter
+        {
+            private static class Expression
+            {
+                public const string nullableSectionValue = "section?.Value";
+                public const string sectionKey = "section.Key";
+                public const string sectionValue = "section.Value";
+            }
+
+            private static class GlobalName
+            {
+                public const string Enum = "global::System.Enum";
+                public const string FromBase64String = "global::System.Convert.FromBase64String";
+                public const string IConfiguration = "global::Microsoft.Extensions.Configuration.IConfiguration";
+                public const string Int32 = "int";
+                public const string IServiceCollection = "global::Microsoft.Extensions.DependencyInjection.IServiceCollection";
+                public const string Object = "object";
+                public const string String = "string";
+            }
+
+            private static class KeyWord
+            {
+                public const string @default = nameof(@default);
+                public const string @null = nameof(@null);
+            }
+
+            private readonly SourceProductionContext _context;
+            private readonly SourceGenerationSpec _generationSpec;
+
+            private readonly Queue<TypeSpec> _privateBindCoreMethodGen_QueuedTypes = new();
+
+            private readonly HashSet<TypeSpec> _internalBindMethodGen_ProcessedTypes = new();
+            private readonly HashSet<TypeSpec> _privateBindCoreMethodGen_ProcessedTypes = new();
+
+            // Postfix for stringValueX variables used to save config value indexer
+            // results e.g. if (configuration["Key"] is string stringValue0) { ... }
+            private int _parseValueCount;
+
+            private readonly SourceWriter _writer = new();
+
+            public Emitter(SourceProductionContext context, SourceGenerationSpec generationSpec)
+            {
+                _context = context;
+                _generationSpec = generationSpec;
+            }
+
+            public void Emit()
+            {
+                _writer.WriteLine(@"// <auto-generated/>
+#nullable enable
+
+using System.Linq;
```

Comment:
>Linq [](http://example.com/codeflow?start=13&length=4)

we'll use Linq in the emitted code? I'll try to look at which places we do that.

Reply:
https://github.com/dotnet/runtime/pull/82179#discussion_r1134665374